scan-command/src/main/java/io/ballerina/scan/internal/ProjectAnalyzer.java

```diff
@@ -69,68 +116,167 @@ private Consumer<DocumentId> analyzeDocument(Module module, ScannerContextImpl s
+            List<URL> jarUrls = pluginDesc.dependencies().stream().map(dependency -> {
+                        try {
+                            return Path.of(dependency.getPath()).toUri().toURL();
+                        } catch (MalformedURLException ex) {
+                            throw new RuntimeException(ex);
+                        }
+            }).toList();
 
-        List<Issue> externalIssues = new ArrayList<>(scannerContextList.size());
-        for (ScannerContext scannerContext : scannerContextList) {
-            ReporterImpl reporter = (ReporterImpl) scannerContext.getReporter();
-            externalIssues.addAll(reporter.getIssues());
+            URLClassLoader ucl = URLClassLoader.newInstance(jarUrls.toArray(new URL[0]),
+                    this.getClass().getClassLoader());
+            InputStream resourceAsStream = ucl.getResourceAsStream(RULES_FILE);
+            if (resourceAsStream == null) {
+                continue;
+            }
+
+            String output;
+            try (BufferedReader reader = new BufferedReader(new InputStreamReader(resourceAsStream,
+                    StandardCharsets.UTF_8))) {
+                output = reader.lines().collect(Collectors.joining());
+            } catch (IOException ex) {
+                throw new RuntimeException(ex);
+            }
+
+            String fqn = pluginDesc.plugin().getClassName();
+            List<Rule> externalRules = new ArrayList<>();
+            Gson gson = new GsonBuilder().setPrettyPrinting().create();
+            JsonElement element = gson.fromJson(output, JsonElement.class);
+            if (!element.isJsonArray()) {
+                outputStream.println("Invalid JSON format for " + RULES_FILE + " in plugin '" + pluginName + "'.");
+                outputStream.println("Expected an array, but found:");
+                outputStream.println(gson.toJson(element));
+                return Optional.empty();
+            }
+
+            JsonArray ruleArray = element.getAsJsonArray();
+            for (JsonElement rule : ruleArray) {
+                JsonObject ruleObject = rule.getAsJsonObject();
+                if (!isValidRule(ruleObject)) {
+                    outputStream.println("Invalid JSON format for rule object in plugin '" + pluginName + "'.");
+                    outputStream.println("Rule object should contain numeric id, ruleKind and description, but found:");
+                    outputStream.println(gson.toJson(ruleObject));
+                    return Optional.empty();
+                }
+
+                String kind = ruleObject.get(RULE_KIND).getAsString();
+                RuleKind ruleKind = switch (kind) {
+                    case BUG -> RuleKind.BUG;
+                    case VULNERABILITY -> RuleKind.VULNERABILITY;
+                    case CODE_SMELL -> RuleKind.CODE_SMELL;
+                    default -> null;
+                };
```

Comment:
Extract this into a separate method

Reply:
This change will not be done in preference to https://github.com/ballerina-platform/static-code-analysis-tool/pull/15#discussion_r1637626715